src/material/schematics/ng-update/migrations/legacy-components-v15/index.ts

```diff
@@ -7,7 +7,111 @@
  */
 
 import {Migration, TargetVersion} from '@angular/cdk/schematics';
+import * as ts from 'typescript';
 
 export class LegacyComponentsMigration extends Migration<null> {
   enabled = this.targetVersion === TargetVersion.V15;
+
+  override visitNode(node: ts.Node): void {
+    if (ts.isImportDeclaration(node)) {
+      this._handleImportDeclaration(node);
+      return;
+    }
+    if (this._isDestructuredAsyncImport(node)) {
+      this._handleDestructuredAsyncImport(node);
+      return;
+    }
+    if (this._isImportExpression(node)) {
+      this._handleImportExpression(node);
+      return;
+    }
+  }
+
+  /** Handles updating the named bindings of awaited @angular/material import expressions. */
+  private _handleDestructuredAsyncImport(
+    node: ts.VariableDeclaration & {name: ts.ObjectBindingPattern},
+  ): void {
+    for (let i = 0; i < node.name.elements.length; i++) {
+      const n = node.name.elements[i];
+      const name = n.propertyName ? n.propertyName : n.name;
+      if (ts.isIdentifier(name)) {
+        const oldExport = name.escapedText.toString();
+        const suffix = oldExport.slice(3);
```

Comment:
Nit: Optional suggestion for cleaner code (not worth micro-optimizing) -- same below

```suggestion
        const suffix = oldExport.slice("Mat".length);
```